com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HDEditorUtils.cs

```diff
@@ -293,6 +293,29 @@ internal static void HandlePrefixLabelWithIndent(Rect totalPosition, Rect labelP
             labelPosition.x += EditorGUI.indentLevel * 15;
             EditorGUI.HandlePrefixLabel(totalPosition, labelPosition, label);
         }
+
+        /// <summary>
+        /// Like EditorGUI.IndentLevelScope but this one will also indent the override checkboxes.
+        /// </summary>
+        internal class IndentScope : GUI.Scope
```

Comment:
Note:
GUIScope : GUI.Scope is a class and as such should produce Garbage Collection at editor frame. 

To remove it you can simply use a struct that implement IDisposable. It will have same benefits but will be all on the stack and will not produce any GC. (in this case IDisposable.Dispose will be the CloseScope())

IMO you should keep this scope as you do the QualityScope.

Also you are not using base CloseScope and base constructor. So perhaps you can use a GUI.Scope instead?

Reply:
Indeed, I originally implemented this scope as a struct using IDisposable (due to the concern for GC like you mention), however doing so produced a set of errors that seem to stem from the usage of GUILayout/EditorGUILayout. Because of that, I followed the pattern of certain builtin scopes in `EditorGUI` that inherit `GUI.Scope`.

After some digging, I'm still not sure exactly what about the scope's implementation makes it prohibitive to use in a normal struct.